src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -181,8 +181,14 @@ def put(
 
                 if result.get("regenerate"):
                     if request.access.has_scope("member:admin"):
-                        member.regenerate_token()
-                        member.save()
+                        region_outbox = None
+                        with transaction.atomic():
+                            member.regenerate_token()
+                            member.save()
+                            region_outbox = member.outbox_for_update()
+                            region_outbox.save()
+                        if region_outbox:
+                            region_outbox.drain_shard(max_updates_to_drain=10)
```

Comment:
Does this need to be synchronous? The attributes being updated here aren't stored on the mapping table.

Reply:
I might introduce a token column to the org member mapping table.